Tests/SwiftProtobufTests/Test_OneodFields_Decoding.swift

```diff
@@ -0,0 +1,72 @@
+// Test/Sources/TestSuite/Test_OneofFields_Decoding.swift
+//
+// Copyright (c) 2014 - 2024 Apple Inc. and the project authors
+// Licensed under Apache License v2.0 with Runtime Library Exception
+//
+// See LICENSE.txt for license information:
+// https://github.com/apple/swift-protobuf/blob/main/LICENSE.txt
+//
+// -----------------------------------------------------------------------------
+///
+/// Exercises the dedcode for primitive fields within a oneof.
+///
+// -----------------------------------------------------------------------------
+
+import Foundation
+import XCTest
+import SwiftProtobuf
+
+final class Test_OneofFields_Decoding: XCTestCase {
```

Comment:
There's already Test_Merge and Test_ParsingMerge, so I'd suggest extending one of those (maybe they should merge).

We also want to make sure everything is covered
- non repeating fields of pod types override
- non repeating fields of message types recursively merge
- repeating fields append
- oneof fields for pod overrides to the new value 
- oneof fields for a message merges
- there's likely cases with extension fields also (repeating/non repeating, pod/message)

I also don't think you need the custom decoder here, just create a two messages, and then use the merge api on the binary data from the second message.  There's some examples in the other files.